flutter/lib/src/navigation/frame_tracker.dart

```diff
@@ -0,0 +1,70 @@
+import 'dart:ui';
+import 'package:flutter/widgets.dart';
+import 'package:sentry/sentry.dart';
+
+class FrameTracker {
+  FrameTracker({
+    this.slowFrameThreshold = const Duration(milliseconds: 16),
+    this.frozenFrameThreshold = const Duration(milliseconds: 500),
+    required WidgetsBinding binding,
+  }) : _binding = binding;
+
+  final WidgetsBinding _binding;
+
+  SingletonFlutterWindow get window => _binding.window;
+
+  final Duration slowFrameThreshold;
+  final Duration frozenFrameThreshold;
+
+  DateTime? _timeStamp;
+
+  var _frameCount = 0;
+  var _slowCount = 0;
+  var _frozenCount = 0;
+
+  void _frameCallback(Duration _) {
+    final timeStamp = _timeStamp;
+    if (timeStamp == null) {
+      return;
+    }
+
+    // postFrameCallbacks are called just once,
+    // so we have to add it each frame.
+    _binding.addPostFrameCallback(_frameCallback);
```

Comment:
Maybe a ticker is a better candidate than this? See https://api.flutter.dev/flutter/scheduler/Ticker-class.html